app/scripts/modules/core/src/pipeline/config/parameters/Parameters.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { every } from 'lodash';
+import { arrayMove, SortableContainer, SortableElement, SortEnd } from 'react-sortable-hoc';
+
+import { HelpField } from 'core/help';
+
+import { IParameter, IParameterProps, Parameter } from './Parameter';
+
+export interface IParametersState {
+  allParametersPinned: boolean;
+}
+
+export interface IParametersProps {
+  parameters: IParameter[];
+  pipelineName: string;
+  addParameter: () => void;
+  removeParameter: (index: number) => void;
+  updateParameter: (index: number, changes: { [key: string]: any }) => void;
+  updateAllParameters: (parameters: IParameter[]) => void;
+}
+
+export class Parameters extends React.Component<IParametersProps, IParametersState> {
+  constructor(props: IParametersProps) {
+    super(props);
+    this.state = {
+      allParametersPinned: false,
+    };
+  }
+
+  public componentDidMount() {
+    this.setPinAllParametersState();
+  }
+
+  private setPinAllParametersState = (): void => {
+    this.setState({ allParametersPinned: every(this.props.parameters || [], { pinned: true }) });
+  };
+
+  private togglePins = (): void => {
+    const parameters = this.props.parameters.slice(0);
+    const { allParametersPinned } = this.state;
+    parameters.forEach(param => {
+      param.pinned = !allParametersPinned;
+    });
+    this.props.updateAllParameters(parameters);
+    this.setPinAllParametersState();
+  };
+
+  private addParameter = (): void => {
+    this.props.addParameter();
+    this.setPinAllParametersState();
+  };
+
+  private removeParameter = (index: number): void => {
+    this.props.removeParameter(index);
+    this.setPinAllParametersState();
+  };
+
+  private updateParameter = (index: number, changes: { [key: string]: any }): void => {
+    this.props.updateParameter(index, changes);
+    this.setPinAllParametersState();
+  };
+
+  private handleSortEnd = (sortEnd: SortEnd): void => {
+    const parameters = arrayMove(this.props.parameters.slice(0), sortEnd.oldIndex, sortEnd.newIndex);
+    this.props.updateAllParameters(parameters);
+  };
+
+  public render(): JSX.Element {
+    const { parameters, pipelineName } = this.props;
+    const { allParametersPinned } = this.state;
+    return (
+      <>
+        <SortableParameters
+          parameters={parameters}
+          allParametersPinned={allParametersPinned}
+          setPinAllParametersState={this.setPinAllParametersState}
+          togglePins={this.togglePins}
+          removeParameter={this.removeParameter}
+          updateParameter={this.updateParameter}
+          isMultiple={parameters.length > 1}
+          onSortEnd={this.handleSortEnd}
+          lockAxis={'y'}
+          useDragHandle={true}
```

Comment:
Does it make sense to move these props (lockAxis, useDragHandle) into `SortableParameters` below directly? They seem unlikely to change and it feels like maybe that's the more appropriate spot for them.

Reply:
`SortableContainer` returns a component class, I would have to create an additional wrapper around `SortableParameters` so that it can default those sortable properties.